### PR TITLE
rbac: Create store methods for roles and permissions (Part 2)

### PR DESCRIPTION
**Pull request by BolajiOlajide**
Part 2 [#45456](https://github.com/sourcegraph/sourcegraph/issues/45456)

This adds store methods for user roles.

## Test plan

Added unit tests for store methods

### CODE REVIEW

File: internal/database/user_roles.go [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+
+func (r *userRoleStore) GetByRoleID(ctx context.Context, opts GetUserRoleOpts) ([]*types.UserRole, error) {
+	role, err := RolesWith(r).GetByID(ctx, GetRoleOpts{
+		ID: opts.RoleID,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	urs := make([]*types.UserRole, 0, 20)
```
Comment (eseliger) [inline internal/database/user_roles.go:184]: why initialize with 20 by default?

File: internal/database/user_roles.go [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+
+func (e *UserRoleNotFoundErr) Error() string {
+	return fmt.Sprintf("user role for user %d and role %d not found", e.UserID, e.RoleID)
+}
+
+func (e *UserRoleNotFoundErr) NotFound() bool {
+	return true
+}
+
+func (r *userRoleStore) Remove(ctx context.Context, opts RemoveUserRoleOpts) error {
```
Comment (eseliger) [inline internal/database/user_roles.go:126]: I think we usually go by `Delete` vs `Remove`

**Review by eseliger — APPROVED** [review]
cool stuff!

File: internal/database/user_roles.go (hunk 1 above)
Comment (BolajiOlajide) [reply internal/database/user_roles.go:184]: This is an oversight. I'll fix it.

File: internal/database/user_roles.go (hunk 2 above)
Comment (BolajiOlajide) [reply internal/database/user_roles.go:126]: Gotcha!

**Pull request merged**